src/main/java/org/truffleruby/core/proc/ProcNodes.java

```diff
@@ -132,6 +139,11 @@ protected Shape getProcShape() {
         protected RubyClass metaClass(RubyProc object) {
             return object.getMetaClass();
         }
+
+        @TruffleBoundary
+        private SourceSection getSection() {
+            return getContext().getCallStack().getTopMostUserSourceSection(getEncapsulatingSourceSection());
```

Comment:
Proc#new is always a core method, so its SourceSection is always skipped, so we should use `getTopMostUserSourceSection()` here (no arguments). That also removes the need for a helper method.

Reply:
Good catch, fixed :+1: